port/linux/ipadapter.c

```diff
@@ -899,6 +892,50 @@ process_event(ip_context_t *dev, fd_set *rdfds, fd_set *wfds)
   return 0;
 }
 
+#ifdef OC_DYNAMIC_ALLOCATION
+static int 
+fds_count(const fd_set *sourcefds) { 
+  int rfd_count = 0;
+  for (int i = 0; i < FD_SETSIZE; i++) { //// TODO: optimize
+    if (FD_ISSET(i, sourcefds)) {
+      rfd_count++;
+    }
+  }
+  return rfd_count;
+}
+
+static int 
+pick_random_fd(const fd_set *sourcefds, int fd_count) {
+  int random_rfd = oc_random_value() % fd_count;
+  for (int i = 0; i < FD_SETSIZE; i++) { // TODO: optimize FD_SETSIZE
+    if (FD_ISSET(i, sourcefds)) {
+      if (--fd_count == random_rfd) {
+        return i;
+      }
+    }
+  }
+  return -1;
+}
+
+static int 
+remove_random_fds(fd_set *rdfds, int rfds_count, int remove_count) {
```

Comment:
@jkralik what's the advantage of closing a random fd instead of the first one for example?